python/lsst/ap/association/transformDiaSourceCatalog.py

```diff
@@ -85,10 +98,38 @@ class TransformDiaSourceCatalogTask(TransformCatalogBaseTask):
 
     ConfigClass = TransformDiaSourceCatalogConfig
     _DefaultName = "transformDiaSourceCatalog"
+    RunnerClass = pipeBase.ButlerInitializedTaskRunner
 
-    def __init__(self, **kwargs):
+    def __init__(self, initInputs, **kwargs):
         super().__init__(**kwargs)
         self.funcs = self.getFunctors()
+        self.inputSchema = initInputs['diaSourceSchema'].schema
+        self._create_bit_pack_mappings()
+
+    def _create_bit_pack_mappings(self):
+        """Setup all flag bit packings.
+        """
+        self.bit_pack_columns = []
+        with open(self.config.flagMap) as yaml_stream:
+            table_list = list(yaml.safe_load_all(yaml_stream))
+            for table in table_list:
+                if table['tableName'] == 'DiaSource':
+                    self.bit_pack_columns = table['columns']
+                    break
```

Comment:
Do you need the `for` loop here, since you're just iterating through until you find the table with the correct name?

Reply:
Considering the yaml file could have flags for different tables (DiaObject, SSObject) and no enforced ordering, it makes sense to just go through the named tables and load the one we want.